core/iwasm/common/wasm_memory.c

```diff
@@ -233,6 +248,85 @@ wasm_runtime_create_shared_heap(SharedHeapInitArgs *init_args)
     return NULL;
 }
 
+WASMSharedHeap *
+wasm_runtime_chain_shared_heaps(WASMSharedHeap *head, WASMSharedHeap *body)
+{
+    WASMSharedHeap *cur;
+    bool heap_handle_exist = head->heap_handle != NULL;
+
+    if (!head || !body) {
+        LOG_WARNING("Invalid shared heap to chain.");
+        return NULL;
+    }
+
```

Comment:
>At most one shared heap in shared heap chain can be dynamically allocated, the rest have to be the pre-allocated shared heap.

might need something to safeguard this⬆️ from intentional or unintentional mistakes made by the caller.